rpc/server.go

```diff
@@ -116,6 +110,64 @@ func marshalResponse(response any, log *zerolog.Logger) []byte {
 	return responseData
 }
 
+type validationResult struct {
+	Error  []byte
+	Method string
+	Id     uint64
+}
+
+func validateRequest(requestData []byte, logger *zerolog.Logger) validationResult {
+	var request map[string]interface{}
+	vr := validationResult{}
+	err := json.Unmarshal(requestData, &request)
+	if err != nil {
+		vr.Error = marshalResponse(unexpectedRequestUnmarshalError, logger)
+		return vr
+	}
+
+	// jsonrpc spec says id can be a string, number.
+	// We only support numbers.
+	// When golang unmarshals JSON into an interface value, float64 is used for numbers.
+	requestId := request["id"]
+	if !assertType[float64](request["id"]) {
+		vr.Error = marshalResponse(invalidRequestError, logger)
+		return vr
+	}
+
+	// is the id an integer?
+	fRequestId := requestId.(float64)
```

Comment:
Is this not repeating the work done on L132 ?

Reply:
I ended up removing `assertType` in f18992f7